tests/Tests/100__deploy/100__installation/105__prometheus.robot

```diff
@@ -0,0 +1,84 @@
+
+*** Test Cases ***
+Verify Prometheus Is Shipped And Enabled Within ODS
+    [Documentation]    Verify Prometheus Is Shipped And Enabled Within ODS
+    [Tags]    Sanity
+    ...       Tier1
+    ...       ODS-232
+    @{prometheus_pods_info} =    Fetch Prometheus Pods Info
+    @{prometheus_deployment_info} =    Fetch Prometheus Deployments Info
+    @{prometheus_services_info} =    Fetch Prometheus Services Info
+    @{prometheus_routes_info} =    Fetch Prometheus Routes Info
+    @{prometheus_replicasets_info} =    Fetch Prometheus ReplicaSets Info
+    OpenShift Resource Field Value Should Be Equal As Strings    status.phase    Running    @{prometheus_pods_info}
+    OpenShift Resource Field Value Should Be Equal As Strings    status.containerStatuses[0].name    alertmanager    @{prometheus_pods_info}
+    OpenShift Resource Field Value Should Be Equal As Strings    status.containerStatuses[1].name    alertmanager-proxy    @{prometheus_pods_info}
+    OpenShift Resource Field Value Should Be Equal As Strings    status.containerStatuses[2].name    oauth-proxy    @{prometheus_pods_info}
+    OpenShift Resource Field Value Should Be Equal As Strings    status.containerStatuses[3].name    prometheus    @{prometheus_pods_info}
+    OpenShift Resource Field Value Should Be Equal As Strings    status.conditions[2].status    True    @{prometheus_pods_info}
+    OpenShift Resource Field Value Should Be Equal As Strings    status.replicas    1    @{prometheus_deployment_info}
+    OpenShift Resource Field Value Should Be Equal As Strings    status.readyReplicas    1    @{prometheus_deployment_info}
+    OpenShift Resource Field Value Should Be Equal As Strings    spec.ports[0].name    https    @{prometheus_services_info}
+    OpenShift Resource Field Value Should Be Equal As Strings    spec.ports[0].port    9091    @{prometheus_services_info}
+    OpenShift Resource Field Value Should Be Equal As Strings    spec.ports[0].protocol    TCP    @{prometheus_services_info}
+    OpenShift Resource Field Value Should Be Equal As Strings    spec.ports[0].targetPort    https    @{prometheus_services_info}
+    OpenShift Resource Field Value Should Match Regexp    spec.clusterIP    ^((25[0-5]|(2[0-4]|1[0-9]|[1-9]|)[0-9])(\.(?!$)|$)){4}$
+    ...    @{prometheus_services_info}
+    OpenShift Resource Field Value Should Be Equal As Strings    status.readyReplicas    1    @{prometheus_replicasets_info}
+    OpenShift Resource Field Value Should Be Equal As Strings    status.replicas    1    @{prometheus_replicasets_info}
+    OpenShift Resource Field Value Should Be Equal As Strings    spec.port.targetPort    https    @{prometheus_routes_info}
+    OpenShift Resource Field Value Should Be Equal As Strings    spec.to.name    prometheus    @{prometheus_routes_info}
+    OpenShift Resource Field Value Should Match Regexp    spec.host    ^(prometheus-redhat-ods-monitoring.*)    @{prometheus_routes_info}
+
+*** Keywords ***
+Fetch Prometheus Pods Info
```

Comment:
are these keywords basically the same as the ines used for Grafana tests? if yes, it may be worth to create generic keywords

Reply:
I'm not sure that a generic keyword could be extracted. Although the verifications that we are doing right now might look similar, under the hood, the microservices that they implement are different, and sooner or later, we'll find that some fields are available in one microservice but not in the other, or that some values have different validations.

Reply:
It the same keywords are needed somewhere else they could be in ODS.robot. Right now I see them here just fine

Reply:
for example, `Fetch Prometheus Services Info` is the same as `Fetch Alert Manager Services Info` and `Fetch Dashboard Services.` So maybe a generic kw named `Fetch Service Info` would work.